domains/pallets/executive/src/lib.rs

```diff
@@ -509,18 +485,7 @@ where
                 <frame_system::Pallet<ExecutiveConfig>>::note_applied_extrinsic(&r, dispatch_info);
                 Ok(Err(err))
             }
-        };
-
-        // TODO: Critical!!! https://github.com/paritytech/substrate/pull/10922#issuecomment-1068997467
```

Comment:
Has the bug or weird behaviour that required this access been resolved? Do we have tests for it?

Reply:
We do not much care about this here since storage roots are computed within runtime but rather once the WASM function itself is completed. Hence removed this.

I did find something else while looking into these but need to confirm my suspicion. Its not related to these changes anyway.